library/waku_thread/inter_thread_communication/request.nim

```diff
@@ -0,0 +1,21 @@
+
+# This file contains the base message request type that will be handled
+# by the Waku Node thread.
+
+import
+  std/json,
+  stew/results
+import
+  chronos
+import
+  ../../../waku/v2/node/waku_node,
+  ../waku_thread
+
+type
+  InterThreadRequest* = ref object of RootObj
```

Comment:
as noted elsewhere, this should not be a `ref object` - rather, the easiest thing to use is a json blob that gets allocated with `allocShared` or the helper you created for it (alloc)